sdk/core/corehttp/corehttp/runtime/policies/_base.py

```diff
@@ -82,8 +85,8 @@ def on_request(self, request: PipelineRequest[HTTPRequestType]) -> Union[None, A
 
     def on_response(
         self,
-        request: PipelineRequest[HTTPRequestType],
-        response: PipelineResponse[HTTPRequestType, HTTPResponseType],
+        request: PipelineRequest[SansIOHTTPRequestType_contra],
+        response: PipelineResponse[SansIOHTTPRequestType_contra, SansIOHTTPResponseType_contra],
```

Comment:
Hah, first time seeing contravariant! I'm not super familiar with the world of generics/covariance/contravariance, but do you know what about `SansIOHttpPolicy` causes it to require contravariant TypeVars, while `HttpPolicy` seems fine with invariant TypeVars?

Reply:
So from what I learned [[here](https://peps.python.org/pep-0544/#overriding-inferred-variance-of-protocol-classes)] and from discussing with Krista, generic protocols cannot be declared as invariant. When the generic type parameters are used in the output, they're covariant. When used in the input, they're contravariant.

In the HTTPPolicy, the output type PipelineResponse uses a [generic type parameter that is covariant](https://github.com/Azure/azure-sdk-for-python/blob/5e0d8882f0d0038a6dd2358316f2a4167d5ae5a9/sdk/core/corehttp/corehttp/runtime/pipeline/__init__.py#L30). So, we didn't need to explicitly make it covariant here.

Reply:
no longer needed